src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -369,6 +379,9 @@ abstract class AbstractCheckTask(
                 .build())
         } finally {
             RuleMetric.decrementActiveRule(type())
+            if (ruleConfiguration.ruleId != 0L) {
+                resultsStorage.putResult(ruleConfiguration.ruleId, EventStatus.forNumber(refs.rootEvent.status.ordinal))
```

Comment:
Why do you take the root event's status? It almost always will be PASSED. We need to check all produced events and find if any of them has FAILED status. If it is the case the rule has failed status. Otherwise, it passed